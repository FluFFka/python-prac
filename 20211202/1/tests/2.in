class nn(metaclass=dump):
    def __init__(self, a, aa=3):
        self.a = aa
        self.aa = a
    def check_eq(self):
        return self == 3.+10.j
    def __eq__(self, other):
        print(self.a, other.imag, self.aa, other.real)
        return self.a == other.imag and self.aa == other.real

n = nn(1)
k = nn(2, aa=4)
l = nn(1)
m = nn(a=3., aa=10.)
print(n.check_eq())
print(k.check_eq())
print(l.check_eq())
print(m.check_eq())
n1 = nn(3, aa=1)
