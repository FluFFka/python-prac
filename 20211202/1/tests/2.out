__init__: (1,), {}
__init__: (2,), {'aa': 4}
__init__: (1,), {}
__init__: (), {'a': 3.0, 'aa': 10.0}
check_eq: (), {}
__eq__: ((3+10j),), {}
3 10.0 1 3.0
False
check_eq: (), {}
__eq__: ((3+10j),), {}
4 10.0 2 3.0
False
check_eq: (), {}
__eq__: ((3+10j),), {}
3 10.0 1 3.0
False
check_eq: (), {}
__eq__: ((3+10j),), {}
10.0 10.0 3.0 3.0
True
__init__: (3,), {'aa': 1}
